### PR TITLE
workaround for incorrect "export is stuck" message when no total count available

### PR DESCRIPTION
**Pull request by czue**

### CODE REVIEW

**Review by snopoke — APPROVED** [review]
small comments but nothing blocking

File: tests/test_commcare_hq_client.py [hunk 1]
```diff
@@ -129,6 +153,9 @@ def test_repeat_limit(self):
                            match="Requested resource '/fake/uri' 10 times with same parameters"):
             self._test_iterate(FakeRepeatedDateCaseSession(), get_paginator('case', 2), 2, [1, 2])
 
+    def test_message_log(self):
+        self._test_iterate(FakMessageLogSession(), get_paginator('messaging-event', 2), 2, [1, 2])
```
Comment (snopoke) [inline tests/test_commcare_hq_client.py:157]: :+1:

File: commcare_export/commcare_hq_client.py [hunk 2]
```diff
@@ -162,6 +163,15 @@ def iterate_resource(resource=resource, params=params):
                     else:
                         more_to_fetch = False
 
+                    limit = batch['meta'].get('limit')
+                    repeated_last_page_of_non_counting_resource = (
+                        not got_new_data
+                        and total_count == 'unknown'
```
[the remark this answers is not in the document]
Comment (czue) [reply commcare_export/commcare_hq_client.py]: https://github.com/dimagi/commcare-export/pull/180/commits/9908562e81dee72321c902c13edac319ccf6f17c

File: commcare_export/commcare_hq_client.py [hunk 3]
```diff
@@ -162,6 +163,15 @@ def iterate_resource(resource=resource, params=params):
...
+                        and (limit and len(batch['objects']) < limit)
+                    )
+                    if more_to_fetch and repeated_last_page_of_non_counting_resource:
+                        more_to_fetch = False
```
[the remark this answers is not in the document]
Comment (czue) [reply commcare_export/commcare_hq_client.py]: https://github.com/dimagi/commcare-export/pull/180/commits/c6ee572a14c4e917acffea3c148dd6b4b7726bd1

**Review by snopoke — APPROVED** [review]

**Pull request merged**